activerecord/lib/active_record/connection_adapters/mysql2/database_statements.rb

```diff
@@ -48,23 +48,30 @@ def perform_query(raw_connection, sql, binds, type_casted_binds, prepare:, notif
             # made since we established the connection
             raw_connection.query_options[:database_timezone] = default_timezone
 
-            result = if prepare
```

Comment:
My understanding is that `prepare` is actually a signal that we want to use the statement cache, based on the historical version. But it was interpreted to mean "are prepared statements enabled or not" but that behavior is decided by the legacy [without_prepared_statement?](https://github.com/rails/rails/blob/d5e70917426d843a7b6139b75cc0c3146d69cbd0/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L1174-L1176)